runtime/caml/domain_state.tbl

```diff
@@ -15,9 +15,14 @@
 /**************************************************************************/
 
 DOMAIN_STATE(atomic_uintnat, young_limit)
-/* Minor heap limit. Typically young_limit == young_start, but this field is set
- * by other domains to signal this domain by causing a spurious allocation
- * failure. */
+/* Minor heap limit. Typically [young_start] <= [young_limit] <=
+ * [young_end], but this field can be set atomically to UINTNAT_MAX by
+ * another thread (typically from another domain) in order to
+ * interrupt this domain (by causing an allocation failure). Setting
+ * [young_limit] to UINTNAT_MAX can be done safely at any time
+ * whatsoever by any thread. To avoid races, setting [young_limit] to
+ * anything else than UINTNAT_MAX should only be done via
+ * [caml_reset_young_limit] by the domain itself. */
```

Comment:
I have two questions about this:

1. Why do we advertise to the whole word that a magic value UINTNAT_MAX is used, instead of exposing a function to check if a domain state received an UINTNAT_MAX interruption? (I realize that @kayceesrk asked a similar question in the past.) In `caml_enter_blocking_section` and in interp.c POPTRAP I would expect to see a `caml_domain_was_interrupted(dom)` test instead of mentioning UINTNAT_MAX.

2. `[Cc]aml_check_gc_interrupt` feels weird, it checks for both "allocation limit reached" and an external interrupt, with a name that suggests that it only checks for interrupts. Does this not risk being misused? I understand the call in `caml_check_urgent_gc` and `check_pending_actions`, but I find the uses in interp.c for POPTRAP and CHECK_SIGNALS suspicious. Why would those want to detect statmemprof triggers or out-of-minor-heap situations?

Reply:
1. I consider `domain_state.tbl` internal. I see this as documenting an implementation rather than an interface.

2. UINTNAT_MAX was not originally intended as a special value. This changed a bit when we fixed the loop inside caml_enter_blocking_section where we explicitly check for UINTNAT_MAX now. I agree it would make sense to check only for UINTNAT_MAX inside POPTRAP and CHECK_SIGNALS. It has remained that way for historical reasons. A PR is welcome, or you could just say "don't fix what's not broken".